openeo_udp/eurac_pv_farm_detection/udf_eurac_pvfarm_onnx.py

```diff
@@ -0,0 +1,156 @@
+import sys
+import os
+import requests
+import zipfile
+from typing import Dict
+import xarray as xr
+import numpy as np
+import functools
+
+dependencies_url = "https://s3.waw3-1.cloudferro.com/swift/v1/project_dependencies/onnx_dependencies_1.16.3.zip"
+model_url = "https://s3.waw3-1.cloudferro.com/swift/v1/project_dependencies/EURAC_pvfarm_rf_1_median_depth_15.zip"
+
+# Function to download and extract zip files
+def download_and_extract(url, extract_to):
+    """
+    Downloads a zip file from the given URL and extracts it to the specified directory.
+    """
+    response = requests.get(url, stream=True)
+    zip_path = os.path.join(extract_to, "temp.zip")
+    
+    with open(zip_path, 'wb') as file:
+        file.write(response.content)
+    
+    with zipfile.ZipFile(zip_path, 'r') as zip_ref:
+        zip_ref.extractall(extract_to)
+    
+    os.remove(zip_path)  # Clean up the zip file
+
+# Prepare paths for dependencies and model
+dependencies_dir = "onnx_deps"
+model_dir = "onnx_models"
+
+# Ensure directories exist
+os.makedirs(dependencies_dir, exist_ok=True)
+os.makedirs(model_dir, exist_ok=True)
+
+# Download and extract dependencies and model if not already present
+if not os.listdir(dependencies_dir):
+    download_and_extract(dependencies_url, dependencies_dir)
+    
+if not os.listdir(model_dir):
+    download_and_extract(model_url, model_dir)
```

Comment:
Isn't it possible to encapsulate all this toplevel makedirs/listdir/download_and_extract stuff in a function? If so, then it would make sense to actually move that kind of functionality to the python client in some kind of UDF utility module.

Reply:
I transfer the code to MLconf.py, should we make a seperate issue to tackle this part independently?

Reply:
I will make the code more modular, but maybe its best if we create a seperate issue for this?